src/mcedit2/panels/worldmeta.py

```diff
@@ -0,0 +1,255 @@
+
+
+class WorldMetaPanel(QtGui.QWidget):
+    GENERATOR_TYPES = ["default", "flat", "largeBiomes", "amplified", "customized", 'debug_all_block_states']
+    GAMEMODES = ["Survival", "Creative", "Adventure", "Spectator"]
+    DIFFICULTIES = ["Peaceful", "Easy", "Normal", "Hard"]
+
+    editsDisabled = False
+
+    def __init__(self, editorSession):
+        super(WorldMetaPanel, self).__init__(QtGui.qApp.mainWindow, f=Qt.Tool)
+        self.editorSession = editorSession
+        self.worldMeta = editorSession.worldEditor.adapter.metadata
+        callIcon = QtGui.QIcon(resourcePath("mcedit2/assets/mcedit2/icons/edit_metadata.png"))
+        callButton = QtGui.QAction(callIcon, "Edit World Metadata", self)
+        callButton.setCheckable(True)
+        callButton.triggered.connect(self.toggleView)
+        self._toggleViewAction = callButton
+
+        load_ui("panels/worldmeta.ui", baseinstance=self)
+
+        self.worldNBTEditor.editorSession = self.editorSession
+        self.editorSession.revisionChanged.connect(self.revisionDidChange)
+
+        self.updateNBTTree()
+        self.setupPanel()
+        self.updatePanel()
+
+        centerWidgetInScreen(self)
+
+    def updatePanel(self):
+        self.defaultGamemodeCombo.setCurrentIndex(self.worldMeta.GameType)
+        self.worldDifficultyCombo.setCurrentIndex(self.worldMeta.Difficulty)
+        self.generationTypeCombo.setCurrentIndex(self.GENERATOR_TYPES.index(self.worldMeta.generatorName))
+
+        self.worldNameLineEdit.setText(self.worldMeta.LevelName)
+        self.worldNameLineEdit.setText(self.worldMeta.LevelName)
+        self.generatorSeedLineEdit.setText(unicode(str(self.worldMeta.RandomSeed), encoding='utf-8'))
+        self.generatorOptionsLineEdit.setText(self.worldMeta.generatorOptions)
+
+        self.spawnX.setValue(self.worldMeta.SpawnX)
+        self.spawnY.setValue(self.worldMeta.SpawnY)
+        self.spawnZ.setValue(self.worldMeta.SpawnZ)
+
+        time = self.worldMeta.DayTime + 30000  # On time = 0, it's day 1, 6:00. Day 0, 0:00 is -30000
+        day = time / 24000
+        hour = (time % 24000) / 1000
+        minute = int((time % 1000) / (1000.0 / 60.0))
+
+        self.timeDays.setValue(day)
+        self.timeHours.setValue(hour)
+        self.timeMinutes.setValue(minute)
+
+        self.lockedDifficultyBool.setChecked(bool(self.worldMeta.DifficultyLocked))
+        self.commandsBool.setChecked(bool(self.worldMeta.allowCommands))
+        self.hardcoreBool.setChecked(bool(self.worldMeta.hardcore))
+
+    def setupPanel(self):
+        self.defaultGamemodeCombo.clear
+        self.defaultGamemodeCombo.addItems(self.GAMEMODES)
+        self.defaultGamemodeCombo.currentIndexChanged.connect(self.defaultGamemodeChanged)
+
+        self.worldDifficultyCombo.clear
+        self.worldDifficultyCombo.addItems(self.DIFFICULTIES)
+        self.worldDifficultyCombo.currentIndexChanged.connect(self.worldDifficultyChanged)
+
+        self.generationTypeCombo.clear
```

Comment:
For this combo box, if you want to translate the displayed text you'll need to keep the underlying string constant since it's part of the NBT format. You can use the second argument to `addItem` to associate "user data" with the item in the choice list and fetch the data with `itemData`